### PR TITLE
Adds a workflow for buf actions

### PR DESCRIPTION
**Pull request by smaye81**
This adds integration with buf Github actions `lint`, `breaking`, and `push`.

Note this needs a Buf API token secret created for this repo.  cc @akshayjshah

### CODE REVIEW

File: .github/workflows/buf.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if: github.ref == 'refs/heads/main'
+    needs:
+      - lint
+      - breaking
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bufbuild/buf-setup-action@v1
+        if: success()
+        with:
+          version: 1.5.0
```
Comment (paul-sachs) [inline .github/workflows/buf.yaml:43]: I wonder why we're using 1.5.0 instead of the latest 1.6.0. But otherwise this looks solid.

**Review by paul-sachs — APPROVED** [review]

File: .github/workflows/buf.yaml (hunk 1 above)
Comment (smaye81) [reply .github/workflows/buf.yaml]: Good point.  Updated.

**Comment by bufdev** [comment]
What are we pushing?

**Comment by smaye81** [comment]
> What are we pushing?

This is to push the Eliza proto files to the BSR here:  https://buf.build/bufbuild/eliza

**Comment by bufdev** [comment]
Why are we pushing these? Is this for the FE @paul-sachs ? How did we decide on bufbuild/eliza?

**Comment by paul-sachs** [comment]
> Why are we pushing these? Is this for the FE @paul-sachs ? How did we decide on bufbuild/eliza?

This is where the FE expects them to live yes. We didn't really find any consistent use for where to put these kinds of things. We can move them elsewhere if necessary.

**Comment by smaye81** [comment]
@bufdev are you able to create a token for the BSR in `secrets` for this repo?

**Pull request merged**